test/fake_rsaprov.c

```diff
@@ -222,6 +243,124 @@ static const OSSL_ALGORITHM fake_rsa_sig_algs[] = {
     { NULL, NULL, NULL, NULL }
 };
 
+static OSSL_FUNC_store_open_fn fake_rsa_st_open;
+static OSSL_FUNC_store_settable_ctx_params_fn fake_rsa_st_settable_ctx_params;
+static OSSL_FUNC_store_set_ctx_params_fn fake_rsa_st_set_ctx_params;
+static OSSL_FUNC_store_load_fn fake_rsa_st_load;
+static OSSL_FUNC_store_eof_fn fake_rsa_st_eof;
+static OSSL_FUNC_store_close_fn fake_rsa_st_close;
+
+static const char fake_rsa_scheme[] = "fake_rsa:";
+
+static void *fake_rsa_st_open(void *provctx, const char *uri)
+{
+    unsigned char *storectx = NULL;
+
+    /* First check whether the uri is ours */
+    if (strncmp(uri, fake_rsa_scheme, sizeof(fake_rsa_scheme) - 1) != 0)
+        return NULL;
+
+    storectx = OPENSSL_zalloc(1);
+    TEST_ptr(storectx);
+
+    TEST_info("fake_rsa_open called");
+
+    return storectx;
+}
+
+static const OSSL_PARAM *fake_rsa_st_settable_ctx_params(void *provctx)
+{
+    static const OSSL_PARAM known_settable_ctx_params[] = {
+        OSSL_PARAM_END
+    };
+    return known_settable_ctx_params;
+}
+
+static int fake_rsa_st_set_ctx_params(void *loaderctx,
```

Comment:
Again, shouldn't this function not exist?

Reply:
We call this unconditionally and the test would crash without the function. That is probably wrong however a fix should be a separate PR.